solutions/loading-web-fonts/pages/index.tsx

```diff
@@ -0,0 +1,213 @@
+          completes, if it determines that its actually used on the page. We can
+          ask the browser to start loading them as soon as it discovers the
+          element, using a <Code>link rel=&quot;preload&quot;</Code> element.
+        </Text>
+        <Snippet>{`<link
+  rel="preload"
+  href="/fonts/Inter-Bold.woff2"
+  as="font"
+  type="font/woff2"
+  crossorigin
+/>`}</Snippet>
+        <Text className="border-l-4 pl-4">
+          If we use <Code>Google Fonts</Code> or <Code>Typekit</Code>,{' '}
+          <Link href="https://nextjs.org/docs/basic-features/font-optimization">
+            Next.js Font Optimization
+          </Link>{' '}
+          will automatically add the preconnect link.
+        </Text>
+      </section>
+
+      <section className="flex flex-col gap-3">
+        <Text variant="h2">We are still not covered</Text>
+        <Text>
+          Althought now our fonts are being loaded, they will not be loaded
+          immediatly. We have a timespan between the font being requested and
+          the font being loaded, that may vary depending on the size of the
+          fonts, the network speed and more.
+        </Text>
+        <Text>
+          During this timespan our text will be displayed with a fallback font
+          or with no font at all depending on the <Code>font-display</Code>{' '}
+          strategy we have defined.
+        </Text>
+        <Text>
+          Based on the visual differences between our custom font and the
+          determined fallback font, we may introduce a layout shift (which is an
+          important metric for Google Core Web Vitals) in our application.
+        </Text>
+        <FontMixer fonts={['Arial', 'Times New Roman']}>
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas
+          convallis mauris dui, non placerat urna fringilla at. Suspendisse a
+          orci quis arcu tristique sagittis at sed leo. Nam quis neque dapibus,
+          semper nisl non, lacinia ligula. Duis viverra a nisl ut consectetur.
+          Ut elementum fringilla odio viverra egestas. Morbi aliquet lorem
+          lorem, in suscipit nibh tempus quis. Mauris gravida dapibus odio,
+          vitae interdum lectus pretium ut. Vivamus tincidunt laoreet
+          pellentesque. Praesent tincidunt elementum tempus.
+        </FontMixer>
+        <Text>
+          Fonts with different sizes not only may create a layout shift in its
+          space but could also push or pull content around it creating a bigger
+          shift in the UI.
+        </Text>
+        <Text>
+          One possible solution is having an adjusted fallback font that tries
+          to match the used space of our custom font to reduce the layout shift.
+          You can use{' '}
+          <Link href="https://www.industrialempathy.com/perfect-ish-font-fallback/?font=Inter">
+            this tool
+          </Link>{' '}
+          to find which fallback font might match your custom font.
```

Comment:
You should mention somewhere that this only works well if `line-height` is set to a non-default value.